packages/builtin/array.pony

```diff
@@ -126,16 +126,20 @@ class Array[A] is Seq[A]
     Create an array from a C-style pointer and length. The contents are not
     copied.
     """
-    _size = len
-
-    if alloc > len then
-      _alloc = alloc
+    if ptr.is_null() then
+      _size = 0
+      _alloc = 0
+      _ptr = ptr
     else
-      _alloc = len
+      _size = len
+      if alloc > len then
+        _alloc = alloc
+      else
+        _alloc = len
+      end
+      _ptr = ptr
     end
```

Comment:
Small nitpick, can the `_ptr = ptr` line be moved back out of the if-else?

Reply:
Done